llvm/include/llvm/Analysis/LoopAccessAnalysis.h

```diff
@@ -304,6 +321,11 @@ class MemoryDepChecker {
   /// restrictive.
   uint64_t MaxSafeVectorWidthInBits = -1U;
 
+  /// Maximum number of elements (power-of-2 and non-power-of-2), which do not
+  /// prevent store-load forwarding and safe to operate simultaneously.
+  std::pair<std::optional<uint64_t>, std::optional<uint64_t>>
```

Comment:
Why do we need to distinguish between non-power and power of 2 here, can't we just use the max of both?